ott/core/initializers.py

```diff
@@ -0,0 +1,279 @@
+# Copyright 2022 The OTT Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Sinkhorn initializers."""
+from typing import Optional, Tuple
+
+import jax
+from jax import numpy as jnp
+
+from ott.core.linear_problems import LinearProblem
+from ott.geometry.pointcloud import PointCloud
+
+
+class SinkhornInitializer():
+
+  def init_dual_a(
+      self, ot_problem: LinearProblem, lse_mode: bool = True
+  ) -> jnp.ndarray:
+    """Initialzation for Sinkhorn potential f.
+
+    Args:
+        ot_problem (LinearProblem): OT problem between discrete distributions of size n and m.
+        lse_mode (bool, optional): Return log potential. Defaults to True.
+    Returns:
+        jnp.ndarray: dual potential, array of size n
+    """
+    return self.default_dual_a(ot_problem=ot_problem, lse_mode=lse_mode)
+
+  def init_dual_b(
+      self, ot_problem: LinearProblem, lse_mode: bool = True
+  ) -> jnp.ndarray:
+    """Initialzation for Sinkhorn potential g.
+
+    Args:
+        ot_problem (LinearProblem): OT problem between discrete distributions of size n and m.
+        lse_mode (bool, optional): Return log potential. Defaults to True.
+    Returns:
+        jnp.ndarray:  dual potential, array of size m
+    """
+    return self.default_dual_b(ot_problem=ot_problem, lse_mode=lse_mode)
+
+  def remove_null_weight_potentials(
+      self,
+      ot_problem: LinearProblem,
```

Comment:
Suggestion: would slightly change the API to accept the weights and the dual vector (+ lse_mode) only.

Reply:
done